pkgs/development/libraries/libirc/default.nix

```
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, cmake, qtbase }:
+
+stdenv.mkDerivation rec {
+  pname = "libirc";
+  version = "unstable-2022-10-15";
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeFlags = [
+    "-DQT5_BUILD=1"
+    "-DQt5Core_DIR=${qtbase.dev}/lib/cmake/Qt5Core"
+    "-DQt5Network_DIR=${qtbase.dev}/lib/cmake/Qt5Network"
+  ];
+
+  preFixup = ''
+    mkdir -p $out/libirc/libirc{,client}
+    cp /build/source/libirc/*.h $out/libirc/libirc
+    cp /build/source/libircclient/*.h $out/libirc/libircclient
+  '';
+
+  src = fetchFromGitHub {
+    owner = "grumpy-irc";
+    repo = "libirc";
+    rev = "734082ffffb6d6744070c75587159d927342edea";
+    sha256 = "Qi/YKLlau0rdQ9XCMyreQdv4ctQWHFIoE3YlW6QnbSI=";
+  };
```

```suggestion
  src = fetchFromGitHub {
    owner = "grumpy-irc";
    repo = "libirc";
    rev = "734082ffffb6d6744070c75587159d927342edea";
    sha256 = "Qi/YKLlau0rdQ9XCMyreQdv4ctQWHFIoE3YlW6QnbSI=";
  };

  nativeBuildInputs = [ cmake ];

  cmakeFlags = [
    "-DQT5_BUILD=1"
    "-DQt5Core_DIR=${qtbase.dev}/lib/cmake/Qt5Core"
    "-DQt5Network_DIR=${qtbase.dev}/lib/cmake/Qt5Network"
  ];

  preFixup = ''
    mkdir -p $out/libirc/libirc{,client}
    cp /build/source/libirc/*.h $out/libirc/libirc
    cp /build/source/libircclient/*.h $out/libirc/libircclient
  '';
```
Please don't use /build/ as a absolute path but a relative one instead.